lib/assert.js

```diff
@@ -885,6 +886,64 @@ assert.ifError = function ifError(err) {
   }
 };
 
+assert.match = function match(string, regexp, message) {
+  if (!isRegExp(regexp)) {
```

Comment:
`String.prototype.match` accepts a string with an implicit conversion `new RegExp(obj)`. Would it make sense to keep aligned with `String.prototype.match`?

Reply:
I think it's best for users to coerce the input to a string before passing it through to the API. It's clear what it stands for in that case instead of relying on implicit behavior.